utils/funcexp/func_time_format.cpp

```diff
@@ -152,19 +152,22 @@ string Func_time_format::getStrVal(rowgroup::Row& row, FunctionParm& parm, bool&
     default: isNull = true; return "";
   }
 
-  const string& format = parm[1]->data()->getStrVal(row, isNull);
+  const auto& format = parm[1]->data()->getStrVal(row, isNull);
 
   char* ptr = buf;
 
   for (uint32_t i = 0; i < format.length(); i++)
   {
-    if (format[i] != '%')
-      *ptr++ = format[i];
+    char fi = format.unsafeStringRef()[i];
+    if (fi != '%')
+      *ptr++ = fi;
     else
     {
       i++;
 
-      switch (format[i])
+      fi = format.unsafeStringRef()[i];
```

Comment:
There is a fi assignment duplication here and at line 161.

Reply:
Note the `i++` between them.